File: math/binary_to_decimal/binary_to_decimal.cpp
// C++ program to convert binary to decimal
#include <iostream>
using namespace std;
 
// Function to convert binary to decimal
int binaryToDecimal(int n)
{
    int num = n;
    int dec_value = 0;
 
    // Initializing base value to 1, i.e 2^0
    int base = 1;
 
    int temp = num;
    while (temp) {
        int last_digit = temp % 10;
        temp = temp / 10;
 
        dec_value += last_digit * base;
 
        base = base * 2;
    }
 
    return dec_value;
}
 
// Driver program to test above function
int main()
{
    int num = 10101001;
 
    cout << binaryToDecimal(num) << endl;
}

File: math/binary_to_decimal/Java/BinaryToDecimal.java
import java.util.Scanner;
public class BinaryToDecimal {
    public static void main(String[] args) 
    {
    long bn, dn = 0, j = 1, r;

    System.out.print("Enter a binary number: ");
    Scanner sc = new Scanner(System.in);
    bn = sc.nextLong();

    while (bn != 0) {
        r = bn % 10;
        dn = dn + r * j;
        j = j * 2;
        bn = bn / 10;
    }

    System.out.println("The decimal number is: " + dn);

    }
}